handlers/handlers.go

```diff
@@ -150,62 +119,197 @@ func (d *CatalystAPIHandlersCollection) processUploadVOD(streamName, sourceURL,
+		return
+	}
+	// Start push per each video track
+	info, err := d.StreamCache.Transcoding.Get(streamName)
+	if err != nil {
+		errorOutInternal("LIVE_TRACK_LIST unknown push source", err)
+		return
+	}
+	for i := range tracks { // i is generated name, not important, all info is contained in element
+		if tracks[i].Type != "video" {
+			// Only produce an rendition per each video track, selecting best audio track
+			continue
 		}
-		lines := strings.Split(strings.TrimSuffix(string(payload), "\n"), "\n")
-		if len(lines) < 2 {
-			errors.WriteHTTPBadRequest(w, "Bad request payload", fmt.Errorf("unknown payload '%s'", string(payload)))
-			return
+		destination := fmt.Sprintf("%s/%s__%dx%d.ts?video=%d&audio=maxbps", info.UploadDir, streamName, tracks[i].Width, tracks[i].Height, tracks[i].Index) //.Id)
+		if err := d.MistClient.PushStart(streamName, destination); err != nil {
+			log.Printf("> ERROR push to %s %v", destination, err)
+		} else {
+			d.StreamCache.Transcoding.AddDestination(streamName, destination)
 		}
+	}
+}
 
-		// stream name is the second line in the Mist Trigger payload
-		s := lines[1]
-		// when uploading is done, remove trigger and stream from Mist
-		errT := d.MistClient.DeleteTrigger(s, "PUSH_END")
-		errS := d.MistClient.DeleteStream(s)
-		if errT != nil {
-			errors.WriteHTTPInternalServerError(w, fmt.Sprintf("Cannot remove PUSH_END trigger for stream '%s'", s), errT)
+// This trigger is run whenever an outgoing push stops, for any reason.
+// This trigger is stream-specific and non-blocking. containing data as such:
+//   push ID (integer)
+//   stream name (string)
+//   target URI, before variables/triggers affected it (string)
+//   target URI, afterwards, as actually used (string)
+//   last 10 log messages (JSON array string)
+//   most recent push status (JSON object string)
+func (d *MistCallbackHandlersCollection) Trigger_PUSH_END(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	streamName := "unknown"
+	errorOutInternal := func(where string, err error) {
+		txt := fmt.Sprintf("%s name=%s", where, streamName)
+		// Mist does not respond or handle returned error codes. We do this for correctness.
+		errors.WriteHTTPInternalServerError(w, txt, err)
+		log.Printf("<ERROR LIVE_TRACK_LIST> %s %v", txt, err)
+	}
+
+	payload, err := io.ReadAll(req.Body)
+	if err != nil {
+		errorOutInternal("Cannot read payload", err)
+		return
+	}
+	lines := strings.Split(strings.TrimSuffix(string(payload), "\n"), "\n")
+	if len(lines) != 6 {
+		errors.WriteHTTPBadRequest(w, "Bad request payload", fmt.Errorf("unknown payload '%s'", string(payload)))
+		return
+	}
+	// stream name is the second line in the Mist Trigger payload
+	streamName = lines[1]
+	destination := lines[2]
+	actualDestination := lines[3]
+	pushStatus := lines[5]
+	if yes, _ := isTranscodeStream(streamName); yes {
+		// Following code is responding to transcoding handler: TODO: encapsulate
```

Comment:
1. Could you address this `TODO`?
2. Also, the same comment as before, we need a place in the comments to explain the whole flow, because it's very hard to grasp the whole logic.